app/controllers/users_controller.rb

```diff
@@ -76,4 +75,18 @@ def refresh_providers
       end
     end
   end
+
+  private
+
+  def cached_providers
+    # retrieve the cached list of all providers
+    response = cmr_client.get_providers
+    response.body if response.success?
+  end
+
+  def all_providers
```

Comment:
these provider methods should also have a default response if they are `error?`s, since otherwise we could be comparing the response objects, and in later calling `.map` on `all_providers`